pkg/kubectl/cmd/apply.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/strategicpatch"
+)
+
+// ApplyOptions is the start of the data required to perform the operation.  As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type ApplyOptions struct {
+	Filenames []string
+}
+
+const (
+	apply_long = `Apply a configuration to a resource by filename or stdin.
+
+JSON and YAML formats are accepted.`
+	apply_example = `# Apply the configuration in pod.json to a pod.
+$ kubectl apply -f ./pod.json
+
+# Apply the JSON passed into stdin to a pod.
+$ cat pod.json | kubectl apply -f -`
+	apply_annotation = "k8s-kubectl-apply"
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#label-selector-and-annotation-conventions

Use something like `kubectl.kubernetes.io/config`

See also the annotations used by rolling update:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/rolling_updater.go#L38
https://github.com/kubernetes/kubernetes/blob/77a06cbf0521c6513bc6062cd0671564124c54ef/pkg/kubectl/kubectl.go#L27